public/components/trace_analytics/components/traces/span_detail_table.tsx

```diff
@@ -32,6 +44,154 @@ export interface SpanSearchParams {
+    display: 'Trace ID',
+  },
+  ...(mode !== 'jaeger'
+    ? [
+        {
+          id: 'traceGroup',
+          display: 'Trace group',
+        },
+      ]
+    : []),
+  {
+    id: mode === 'jaeger' ? 'process' : 'serviceName',
+    display: 'Service',
+  },
+  {
+    id: mode === 'jaeger' ? 'operationName' : 'name',
+    display: 'Operation',
+  },
+  {
+    id: mode === 'jaeger' ? 'duration' : 'durationInNanos',
+    display: 'Duration',
+    initialWidth: 100,
+  },
+  {
+    id: mode === 'jaeger' ? 'tag' : 'status.code',
+    display: 'Errors',
+    initialWidth: 100,
+  },
+  {
+    id: 'startTime',
+    display: 'Start time',
+  },
+  {
+    id: mode === 'jaeger' ? 'jaegerEndTime' : 'endTime',
+    display: 'End time',
+  },
+];
+
+const renderCommonCellValue = ({
+  rowIndex,
+  columnId,
+  items,
+  tableParams,
+  expandedRows,
+  toggleRowExpansion,
+  props,
+  flattenedItems,
+  indentationFactor = 0,
+  fullScreenMode = false,
+}: {
+  rowIndex: number;
+  columnId: string;
+  items: any;
+  tableParams: any;
+  expandedRows?: Set<string>;
+  toggleRowExpansion?: (spanId: string) => void;
+  props: SpanDetailTableProps;
+  flattenedItems?: any[];
+  indentationFactor?: number;
+  fullScreenMode?: boolean;
+}) => {
+  const adjustedRowIndex = flattenedItems
+    ? rowIndex
+    : rowIndex - tableParams.page * tableParams.size;
+  const item = flattenedItems ? flattenedItems[rowIndex] : items[adjustedRowIndex];
+
+  if (!item) return '-';
+
+  const value = item[columnId];
+  const indentation = `${(item.level || 0) * indentationFactor}px`;
+  const isRowExpanded = expandedRows?.has(item.spanId);
+
+  if ((value == null || value === '') && columnId !== 'jaegerEndTime') return '-';
+
+  switch (columnId) {
+    case 'tag':
+      return value?.error === true ? (
+        <EuiText color="danger" size="s">
+          Yes
+        </EuiText>
+      ) : (
+        'No'
+      );
+    case 'references':
+      return value.length > 0 ? value[0].spanID : '';
+    case 'process':
+      return value?.serviceName;
+    case 'spanId':
+    case 'spanID':
+      return (
+        <div style={{ paddingLeft: indentation, display: 'flex', alignItems: 'center' }}>
+          {toggleRowExpansion && item.children?.length > 0 ? (
+            <EuiIcon
+              type={isRowExpanded ? 'arrowDown' : 'arrowRight'}
+              onClick={() => toggleRowExpansion(item.spanId)}
+              style={{ cursor: 'pointer', marginRight: 5 }}
+            />
+          ) : (
+            <EuiIcon type="empty" style={{ visibility: 'hidden', marginRight: 5 }} />
+          )}
+          {!fullScreenMode ? (
+            <EuiLink data-test-subj="spanId-link" onClick={() => props.openFlyout(value)}>
+              {value}
+            </EuiLink>
+          ) : (
+            <span>{value}</span>
+          )}
+        </div>
+      );
+    case 'durationInNanos':
+      return `${round(nanoToMilliSec(Math.max(0, value)), 2)} ms`;
```

Comment:
What's with the names here, where the column's id says 'nanos' but the function for the render seems to be showing milliseconds, where it calls a method named 'nanoToMilliSec'?

Reply:
I kept that from the originally implementation which has been in place for 3 years, not sure the original reasoning.
https://github.com/opensearch-project/dashboards-observability/commit/dde958863a48456d2affc5b628614a38346e7088#diff-491203dc294dcf7e9d988e5ba55b77471fccebc3e7d3d62c42dd0b0a6a365599R136